backend/src/Message/TaskCompletedMessageHandler.php

```diff
@@ -24,13 +24,14 @@ public function __invoke(TaskCompletedMessage $message): void
             $this->removeGlobalOldTasks();
         }
 
+        /** @var Task $task */
         $task = $this->entityManager->getRepository(Task::class)->find($message->taskId);
         if (!$task) {
             return;
         }
```

Comment:
Good practice to retrieve the `projectId` correctly from the `Task` object via the `project` relation. This change ensures consistency and reduces the risk of referencing an incorrect or non-existent project ID.